rotkehlchen/db/dbhandler.py

```diff
@@ -3528,7 +3532,18 @@ def get_history_events(self, filter_query: HistoryEventFilterQuery) -> List[Hist
                 self.msg_aggregator.add_error(
                     f'Failed to read history event from database. {str(e)}',
                 )
-        return result
+
+        if filter_query.pagination is not None:
+            no_pagination_filter = deepcopy(filter_query)
+            no_pagination_filter.pagination = None
+            query, bindings = no_pagination_filter.prepare()
```

Comment:
Why are you doing this entire thing? Please just copy the logic that already exists in the code.Check `get_asset_movements` and `get_asset_movements_and_limit_info`.

We have `filter_query.prepare(with_pagination=False)` for a reason.